azurelinuxagent/common/utils/cryptutil.py

```diff
@@ -127,4 +140,14 @@ def bits_to_bytes(self, bits):
                 curr = 0
                 index = 7
         return bytes(byte_array)
-
+    
+    def decrypt_secret(self, encrypted_password, private_key, cache_file, encoding='utf-8'):
+        try:
+            decoded = textutil.b64decode(encrypted_password, encoding)
+            with open(cache_file, "wb") as c:
+                c.write(decoded)
```

Comment:
Is it really necessary to write it to disk just to decrypt?

Reply:
I tried the suggestions before, but none seemed to work as the command had issues reading the binary data off the command even when I tried reading from /dev/stdin.  I eventually gave up and implemented it like this so I could move forward (worst case, it's still encrypted on disk so there's no real security issue... but I had hoped to come back to look at this again. There's another comment with another suggestion for trying to avoid this and I will try it.  

---
In reply to: [192193948](https://github.com/Azure/WALinuxAgent/pull/1171#discussion_r192193948) [](ancestors = 192193948)